src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeleteDatasetVersionCommand.java

```diff
@@ -36,7 +41,7 @@ public DeleteDatasetVersionCommand(DataverseRequest aRequest, Dataset dataset) {
 
     @Override
     protected void executeImpl(CommandContext ctxt) throws CommandException {
-        ctxt.permissions().checkEditDatasetLock(doomed, getRequest(), this);
+        permissionsWrapper.checkEditDatasetLock(doomed, getRequest(), this);
```

Comment:
I've always thought of the `PermissionsWrapper` as UI-only feature that we only call from JSF backing beans but if we are able to use it in commands, that means that the API can benefit from the speed up as well! This is great. My only wonder is if we should `@Inject` `PermissionsWrapper` into individual commands like this. It might be better to weave `PermissionsWrapper` into the "engine` @michbarsinai built. I suppose this could be a future endeavor. I don't want to hold up progress.